doc/vxlan/Overlay ECMP ehancements.md

```diff
@@ -0,0 +1,227 @@
+
+
+### 3.1.1 Behaviour example
+
+| Scenario |     Before state change    |       State change       | Result                 | Description|
+|:--------:|:--------------------------:|:------------------------:|:----------------------:|:-----------|
+| Single primary-secondary switchover. Endpoint list = [A, A’], Primary[A] | NH=[A]  | A went Down  | NH=[A’]  | NH has a single primary endpoint which upon failing is replaced by the single Backup endpoint. |
+| Single primary recovery. Endpoint list = [A, A’] Primary[A] | NH=[A’]  | A is back up  | NH=[A]  | NH has a single backup endpoint which upon recovery of primary is replaced.  |
+| Single primary backup Failure. Endpoint list = [A, A’].  Primary[A]| NH=[A’] A is DOWN | A’ goes Down  | NH=[]  | No active Endpoint results in route being removed.  |
+| Multiple primary backups. Single primary failure.  Endpoint list = [A, B, C, A’, B’, C’]  Primary = [A, B, C] | NH = [A, B, C]  | A goes Down  | NH=[B, C]  | One of the primaries goes down. The others stay active.  |
+| Multiple primary backups. Multiple primary failure. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [B, C]  | B goes Down. A already Down. | NH=[C]  | 2 of the primaries goes down. The 3rd stay active. NH group is updated.  |
+| Multiple primary backups.  All primary failure. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [C]  | C goes Down. A,B already Down  | NH=[A’, B’, C’]  | All the primaries are down. The backup endpoints are added to the NH group.  |
+| Multiple primary backups. Backup Failure. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [A’, B’, C’]  | C’ goes Down. A, B, C already Down  | NH=[A’, B’]  | All the primaries are down. Failure of a backup endpoint shall result in its removal from NH.  |
+| Multiple primary backups. Single primary recovery. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [A’, B’, C’]  | A is Up. B, C still Down  | NH=[A]  | Primary takes precedence and is added to the NH. All the backups are removed.  |
+| Multiple primary backups. Multiple primary recovery. Endpoint list = [A, B, C, A’, A’, B’, C’] Primary = [A, B, C]  | NH = [A]  | A is Up. B, C also come up  | NH=[A, B, C]  | Primary endpoints take precedence and are added to the NH.  |
+ 
+
+## 3.2 Custom monitoring for VTEP liveness detection.
+In the orignal design of Overlay-ECMP, BFD was used for livness detection fo VTEP. But BFD may not be supported by all types of VTEPs. In some cases a user may want to use their own custom protocol for livness detection. This enhacement allows for such a mechanism. The Orchagent creates an entry in the VNET_MONITOR_TABLE in APP DB if it recieves the "monitoring" = "custom" attribute in the VNET_ROUTE_TUNNEL_TABLE entry. The orchaagent then listens on VNET_MONITOR_TABLE in the STATE DB to monitor the livness of the VTEP.
```

Comment:
add a diagram to futher explain this . how the custom monitoring is chosen and that we cannot have both custom and BFD monitoring in the same tunnel route endpoints.

Reply:
done